identity_iota_core/src/rebased/client/read_only.rs

```diff
@@ -70,97 +69,106 @@ impl IdentityClientReadOnly {
-      /// network.
-      ///
-      /// # Notes
-      /// When trying to connect to a local or unofficial network prefer using
-      /// [`IdentityClientReadOnly::new_with_pkg_id`].
-      pub async fn new(iota_client: WasmIotaClient) -> Result<Self, Error> {
-        Self::new_internal(IotaClientAdapter::new(iota_client)?).await
-      }
-    } else {
-      /// Attempts to create a new [`IdentityClientReadOnly`] from a given [`IotaClient`].
-      ///
-      /// # Failures
-      /// This function fails if the provided `iota_client` is connected to an unrecognized
-      /// network.
-      ///
-      /// # Notes
-      /// When trying to connect to a local or unofficial network prefer using
-      /// [`IdentityClientReadOnly::new_with_pkg_id`].
-      pub async fn new(iota_client: IotaClient) -> Result<Self, Error> {
-        Self::new_internal(IotaClientAdapter::new(iota_client)?).await
-      }
-    }
+  /// Attempts to create a new [`IdentityClientReadOnly`] from a given [`IotaClient`].
+  ///
+  /// # Failures
+  /// This function fails if the provided `iota_client` is connected to an unrecognized
+  /// network.
+  ///
+  /// # Notes
+  /// When trying to connect to a local or unofficial network prefer using
+  /// [`IdentityClientReadOnly::new_with_pkg_id`].
+  pub async fn new(
+    #[cfg(target_arch = "wasm32")] iota_client: WasmIotaClient,
+    #[cfg(not(target_arch = "wasm32"))] iota_client: IotaClient,
+  ) -> Result<Self, Error> {
+    let client = IotaClientAdapter::new(iota_client);
+    let network = network_id(&client).await?;
+    Self::new_internal(client, network).await
   }
 
-  async fn new_internal(iota_client: IotaClientAdapter) -> Result<Self, Error> {
-    let network = network_id(&iota_client).await?;
-    let metadata = iota::well_known_networks::network_metadata(&network).ok_or_else(|| {
-      Error::InvalidConfig(format!(
-        "unrecognized network \"{network}\". Use `new_with_pkg_id` instead."
-      ))
-    })?;
-    // If the network has a well known alias use it otherwise default to the network's chain ID.
-    let network = metadata.network_alias().unwrap_or(network);
-
-    let pkg_id = metadata.latest_pkg_id();
-
+  async fn new_internal(iota_client: IotaClientAdapter, network: NetworkName) -> Result<Self, Error> {
+    let chain_id = network.as_ref().to_string();
+    let (network, iota_identity_pkg_id) = {
+      let package_registry = iota::package::identity_package_registry().await;
+      let package_id = package_registry
+        .package_id(&network)
+        .ok_or_else(|| {
+        Error::InvalidConfig(format!(
+          "no information for a published `iota_identity` package on network {network}; try to use `IdentityClientReadOnly::new_with_package_id`"
+        ))
+      })?;
+      let maybe_alias = package_registry
+        .chain_alias(&chain_id)
+        .and_then(|alias| NetworkName::try_from(alias).ok());
```

Comment:
Could we move this to the default case in the match, as it seems to be only used there?